tests/test_api.py

```diff
@@ -1218,6 +1218,20 @@ def test_nullable_datetime_columns(self):
         }
         json.loads(self.app.post("/query", data=json.dumps(query)).data)
 
+    def test_aggregate_timestamp(self):
+        query = {
+            "project": 1,
+            "selected_columns": ["timestamp"],
```

Comment:
You shouldn't have this problem by removing the timestamp column from here.
the groupby clause will automatically add it. Or am I misunderstanding the issue ?

Reply:
This works:
```
{
    "selected_columns": [],
    "aggregations": [
        ["argMax", ["project_id", "timestamp"], "projectid"],
        ["argMax", ["event_id", "timestamp"], "latest_event"]
     ],
    "groupby": ["timestamp"],
    "project": [1],
    "from_date": "2019-11-20T20:02:09",
    "to_date": "2019-11-25T20:02:09",
    "granularity": 3600
}
```

Also I guess you should use one of the time group columns when you want to group by that apply granularity properly.
> time_group_columns={"time": "timestamp", "rtime": "received"},

Reply:
I ran into this issue as the discover query builder/transform layers are including fields in both `groupby` and `selected_columns`. If the fields can be omitted from `selected_columns` I can fix that in the query generation in sentry.

Reply:
I have a better use case for this test.
```
{
    "selected_columns": ["timestamp", "timestamp"],
    "limit": 3,
    "project": [1],
    "from_date": "2019-11-21T01:00:36",
    "to_date": "2019-11-26T01:00:36",
    "granularity": 3600
}
```

So you do not need to stress the fact that groupby transparently adds another column. This will fail as well and as well should be fixed by your test.